contracts/InitContract.json

```diff
@@ -18,10 +30,19 @@
+  "deployedBytecode": "0x60606040526004361061004c576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff16806343d726d614610051578063ec9397fd14610066575b600080fd5b341561005c57600080fd5b610064610100565b005b341561007157600080fd5b6100fe60048080359060200190820180359060200190808060200260200160405190810160405280939291908181526020018383602002808284378201915050505050509190803590602001908201803590602001908080602002602001604051908101604052809392919081815260200183836020028082843782019150505050505091905050610166565b005b60405180807f6465737472756374696e67000000000000000000000000000000000000000000815250600b01905060405180910390207f2844c95bf1b4598da931d527f903501abc60fe0199c65da52d5ce818c6c5e96160405160405180910390a26000ff5b8051825114151561017657600080fd5b73222222222222222222222222222222222222222273ffffffffffffffffffffffffffffffffffffffff1663ec9397fd83836040518363ffffffff167c0100000000000000000000000000000000000000000000000000000000028152600401808060200180602001838103835285818151815260200191508051906020019060200280838360005b8381101561021a5780820151818401526020810190506101ff565b50505050905001838103825284818151815260200191508051906020019060200280838360005b8381101561025c578082015181840152602081019050610241565b50505050905001945050505050600060405180830381600087803b151561028257600080fd5b5af1151561028f57600080fd5b50505050505600a165627a7a72305820ef154045c31652309d0710bf9afba27a87dece9fa20740280c88b5e028b7300e0029",
   "sourceMap": "57:233:0:-;;;;8:9:-1;5:2;;;30:1;27;20:12;5:2;57:233:0;;;;;;;",
   "deployedSourceMap": "57:233:0:-;;;;;;;;;;;;;;;;;;;;;;;;85:203;;8:9:-1;5:2;;;30:1;27;20:12;5:2;85:203:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;177:6;:13;162:4;:11;:28;154:37;;;;;;;;215:42;201:66;;;268:4;274:6;201:80;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;23:1:-1;8:100;33:3;30:1;27:10;8:100;;;99:1;94:3;90:11;84:18;80:1;75:3;71:11;64:39;52:2;49:1;45:10;40:15;;8:100;;;12:14;201:80:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;23:1:-1;8:100;33:3;30:1;27:10;8:100;;;99:1;94:3;90:11;84:18;80:1;75:3;71:11;64:39;52:2;49:1;45:10;40:15;;8:100;;;12:14;201:80:0;;;;;;;;;;;;;;;;;;;;;;;;8:9:-1;5:2;;;30:1;27;20:12;5:2;201:80:0;;;;8:9:-1;5:2;;;45:16;42:1;39;24:38;77:16;74:1;67:27;5:2;201:80:0;;;;85:203;;:::o",
   "source": "pragma solidity ^0.4.21;\n\nimport \"./ProxyContract.sol\";\n\ncontract InitContract {\n    function setValue(bytes32[] keys, bytes32[] values) public {\n        require(keys.length == values.length);\n        ProxyContract(0x2222222222222222222222222222222222222222).setValue(keys, values);\n    }\n}\n",
```

Comment:
Could you also add the source code of the changed contract, please?

Reply:
As we discussed in a previous meeting you wanted to have it only in a json format as the source code is in the json file as well.

Below you'll find the updated versions of both contracts (I can also push them of course):

InitContract.sol:
`pragma solidity ^0.4.21;

import \"./ProxyContract.sol\";

contract InitContract {
    function setValue(bytes32[] keys, bytes32[] values)
    public
    {
        require(keys.length == values.length);
        ProxyContract(0x2222222222222222222222222222222222222222).setValue(keys, values);
    }

    function close()
    public
    {
        emit Status('destructing');
        selfdestruct(0x0);
    }

    event Status(string indexed status);
}`

ProxyContract.sol
`pragma solidity ^0.4.21;

contract ProxyContract {
    function setValue(bytes32[] keys, bytes32[] values) public {
        address initializationContract = 0x8888888888888888888888888888888888888888;
        require(msg.sender == initializationContract);
        require(keys.length == values.length);
        for(uint i = 0; i < keys.length; i++) {
            assembly {
                sstore(mload(add(add(keys, 0x20), mul(i, 0x20))), mload(add(add(values, 0x20), mul(i, 0x20))))
            }
        }
    }

    //https://blog.gnosis.pm/solidity-delegateproxy-contracts-e09957d0f201
    function() public payable {
        bool success = 0x2222222222222222222222222222222222222222.delegatecall(msg.data);
        assembly {
            let mempointer := mload(0x40)
            returndatacopy(mempointer, 0, returndatasize())
            switch success
                case 0 { revert(mempointer, returndatasize()) }
                default { return(mempointer, returndatasize()) }
        }
    }
}`

Reply:
> EDIT: Unfortunately I don't get the code block for the contracts working in this comment.

Could you elaborate, please? What code block isn't working?

> As we discussed in a previous meeting you wanted to have it only in a json format as the source code is in the json file as well.

Yes, I was thinking of adding the high level code under the `source` key of the json though.

Reply:
Alright, this is solved in PR #21.